packages/block-library/src/cover/index.js

```diff
@@ -67,6 +68,12 @@ const blockAttributes = {
 		type: 'string',
 		default: 'image',
 	},
+	focalPoint: {
+		type: 'object',
+	},
+	dimensions: {
+		type: 'object',
+	},
```

Comment:
Why do we need to store the dimensions? Can't we just use percentage in the focal point instead?

Reply:
The focal point picker needs image dimensions for two reasons:

- To define the drag area for the target. The user should only be able to drag the target over the area of the image, and unless the image's aspect ratio precisely matches the focal point picker's area there will be blank space on the left/right or top/bottom. Awareness of the dimensions allows a precise definition of the drag area. This is done here: https://github.com/WordPress/gutenberg/blob/97fcd04d05ce2ae039f891225a6a1fb781edfab2/packages/components/src/focal-point-picker/index.js#L39-L64
- To convert the location of the target to a percentage-based focal point we need to know the dimensions of the original image. This calculation occurs here: https://github.com/WordPress/gutenberg/blob/97fcd04d05ce2ae039f891225a6a1fb781edfab2/packages/components/src/focal-point-picker/index.js#L65-L90

However, I don't love requiring dimensions as a parameter for the component. First of all, it's restrictive - ideally the component just takes the image path and handles the rest. And second, it also creates a deprecation problem for Cover blocks created prior to the Focal Point Picker in which the dimensions are not retained. Ideally, I'd prefer that the Focal Point Picker component determine the dimensions programmatically from the image itself. I'm going to experiment with a few approaches to this and will update the PR shortly.

Reply:
In https://github.com/WordPress/gutenberg/pull/10925/commits/5567d1649dfde7a289362df23aa09e2c884ed109 I've introduced a different approach to getting the image dimensions. The parameters are gone. Instead, the component creates an invisible `<img />` (opacity: 0) and derives the dimensions from that. This way the block using the component need not supply dimensions and the component will work with legacy blocks. Does this seem like a viable approach?